dbt_common/record.py

```diff
@@ -147,17 +152,46 @@ def print_diffs(self) -> None:
 
 
 def get_record_mode_from_env() -> Optional[RecorderMode]:
```

Comment:
Not a blocker for the current PR but a general observation, I think we are basically implementing something similar to pydantic settings functionalities here. We should consider consolidating all env readings(other than the ones that are also flags in click commands) to use it.

Reply:
https://github.com/dbt-labs/dbt-core/issues/10100 addresses this.  Going to leave this as is for now.

Reply:
Sounds good to me!